src/main/java/app/bpartners/api/service/subscription/SubscriptionService.java

```diff
@@ -377,6 +350,40 @@ public Redirection initiateSubscription(
                 .failureUrl(session.getCancelUrl()));
   }
 
+  private Session createSession(
+      LocalDate trialEnd,
+      Customer stripeCustomer,
+      RedirectionStatusUrls redirectionUrls,
+      Subscription subscription,
+      Price newVariableProductPrice,
+      Long billingCycleAnchor,
+      SubscriptionProduct subscriptionProduct)
+      throws StripeException {
+    if (trialEnd.isAfter(temporalUtils.fifthOfNextMonth())) {
+      return stripeSessionFactory.createSessionSetUp(
```

Comment:
No. 

Actually, it must be reveresed, where is the test I've asked for to look at it correctly ?

If it's the 6th April, we can directly create the subscription as the period between 6th April to 5th May is less than natural_billing_cycle (30 days). But when the period before the next billing is > 30 day that's what you should handle.